nxp: imx: change CONFIG_SOC_<name> to match the value

iuliana-prodan opened the pull request:
Change, for audio DSP core, the `SOC_<name>` to match the name

iuliana-prodan committed:
nxp: imx8: change CONFIG_SOC_<name> to match the value

Change, for audio DSP and M7 cores, SOC_<name> to match
the exact soc name.
Update the board files accordingly.

These configs are used in SOF and NXP_HAL, so change
sha for these modules.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan commented:
The update of SOF from `submanifests/optional.yaml` brings some dependencies that are not yet merged in Zephyr (this PR https://github.com/zephyrproject-rtos/zephyr/pull/70219) so the CI fails with:
`ERROR   - sample.audio.sof - integration_platforms - unrecognized platform - imx8ulp_evk/mimx8ud7/adsp`

dbaluta commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -40,7 +40,7 @@ config SOC_MIMX8ML8_ADSP
 	help
 	  Enable support for NXP i.MX 8MPLUS Audio DSP
 
-config SOC_MIMX8MP_M7
```
@iuliana-prodan I would also add some comments saying 

```# i.MX8MP in human speak```

iuliana-prodan replied:
There are some info in the help, see, for example:
```
help
	  Enable support for NXP i.MX 8MPLUS M7 MCU
```
or
```
help
	  Enable support for NXP i.MX 8MPLUS Audio DSP
```

dbaluta replied:
thanks.this will do.

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -40,7 +40,7 @@ config SOC_MIMX8ML8_ADSP
 	  Enable support for NXP i.MX 8MPLUS Audio DSP
 
-config SOC_MIMX8MP_M7
+config SOC_MIMX8ML8_M7
```
Squash all board commits to one commit then a separate commit for both manifest updates so that chance of breaking git bisect is reduced

iuliana-prodan replied:
done.

LaurentiuM1234 reviewed (approved):
LGTM! Might be a good idea to squash all commits into a single one when dependencies are merged. Current layout seems to break git bisect?

iuliana-prodan commented:
Changes since last version:
- update sha for sof in submanifest/optional.yml;
- rebase.

danieldegrasse commented:
@dleach02 can you take a look at this PR, and if all is good with you merge the HAL PR (https://github.com/zephyrproject-rtos/hal_nxp/pull/361)?

dleach02 commented:
HAL_NXP PR merged. Please update manifest

iuliana-prodan commented:
> Changes since last version:
> 
> * update sha for sof in submanifest/optional.yml;
> * rebase.
> 
> Still a dependency on hal_nxp - [zephyrproject-rtos/hal_nxp#361](https://github.com/zephyrproject-rtos/hal_nxp/pull/361)


Changes since last version:

- update sha for nxp_hal in west.yml;
- rebase;
- squash all commits to not break bisect.

All dependencies are merged

LaurentiuM1234 commented:
@nordicjm could you please have another look? Should be good to go at this point.
@nordic-krch can you please help review this?

Thanks!

mmahadevan108 commented:
@nordic-krch can you help review this PR
